addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -162,6 +221,8 @@ PaymentForm.include({
         // first we check that the user has selected a stripe as s2s payment method
         if ($checkedRadio.length === 1 && this.isNewPaymentRadio($checkedRadio) && $checkedRadio.data('provider') === 'stripe') {
             return this._createStripeToken(ev, $checkedRadio);
+        } else if ($checkedRadio.attr('name') === 'pm_id' && !this.isNewPaymentRadio($checkedRadio) && !this.isFormPaymentRadio($checkedRadio) && $checkedRadio.data('provider') === 'stripe') {
```

Comment:
I think we should merge this with 1st condition
`($checkedRadio.length === 1 && (this.isNewPaymentRadio($checkedRadio) || ($checkedRadio.attr('name') === 'pm_id') && $checkedRadio.data('provider') === 'stripe')`